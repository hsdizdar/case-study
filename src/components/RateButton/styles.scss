@import "../../styles/styles";

.rate-button {
    flex: 1;
    align-items: center;
    background-color: transparent;
    border: none;
    border-right: 1px solid $border;
    padding: 10px 0px 10px 0px;

    &:hover {
        cursor: pointer;
    }

    &:focus, &:active {
        outline: none;
        border: none;
    }
    
    &--active {
        flex: 1;
        align-items: center;
        background-color: $rate;
        border: none;
        padding: 10px 0px 10px 0px;

        &:focus, &:active {
            outline: none;
            border: none;
        }
    }
}