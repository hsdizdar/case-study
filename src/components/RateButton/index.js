import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import MatchContext from '../../context/MatchContext';

import './styles.scss';

const RateButton = ({ id, rate, onClick }) => {
  const context = useContext(MatchContext);

  const active = context && context.matches.find((currentMatch) => currentMatch.rateId === id);

  return (
    <button className={active ? 'rate-button--active' : 'rate-button'} onClick={onClick}>
      {rate}
    </button>
  );
};

RateButton.propTypes = {
  id: PropTypes.string,
  rate: PropTypes.string,
  onClick: PropTypes.func,
};

export default RateButton;
