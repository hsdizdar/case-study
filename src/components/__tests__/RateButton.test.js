import React from 'react';
import renderer from 'react-test-renderer';
import RateButton from '../RateButton';


it('renders correctly when there are no items', () => {
    const tree = renderer.create(<RateButton />).toJSON();
    expect(tree).toMatchSnapshot();
});
