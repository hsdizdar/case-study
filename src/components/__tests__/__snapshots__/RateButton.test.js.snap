// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<button
  className="rate-button"
/>
`;
